### DISCUSSION TITLE
심화프로젝트 2주차 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#443] [회고]
# 왜 회고를 해야하나요?
- **원활한 팀 프로젝트 진행을 위해, 회고가 좋은 도구가 되어줄거예요.** 
매주 팀 단위로 회고를 하며 어떤 점이 좋았고, 어떤 점이 아쉬웠고, 아쉬웠던 부분은 어떻게 개선해볼지 함께 이야기해보면 
그 다음주에는 더 나은 프로젝트를 진행할 수 있을거예요. 회고를 통해 함께 성장해보아요 🔥 
- **회고 시간만큼은 솔직한 생각, 마음을 터놓아보기! 😎** 
솔직한 회고를 통해 다음주는 더 즐겁게 프로젝트를 진행할 수 있을거예요.  

# 회고 방법
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/b1f1cc83-7d24-4156-910d-3fc9b6bb7555)
회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다. 
- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
    - Keep: 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
    - Problem: 현재 우리팀이 가지고 있는 문제
    - Try: Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/0NHIhlM9vhYZDrtZa7nJn8/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF?type=whiteboard&node-id=0%3A1&t=p6vXDVEjQntHKaNP-1)을 활용해서, 팀 회고를 진행하셔도 됩니다. 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요. 
- **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡처본을 올려주셔도 돼요.)

# 참고 글
- [KPT 회고란 무엇인가?](https://brunch.co.kr/@jinha0802/35)
- [팀문화의 탄생](https://techblog.woowahan.com/2677/)

**Comment by 9ummy9** [comment]
# 6팀

## Keep 👌
   - 팀미팅 및 전체회의에 성실하게 참여한다.
   - 각자 맡은 일을 성실히 수행한다.
   - 프로젝트 관련 아이디어를 잘 낸다.
   - 의견을 제시할 때 근거와 함께 이야기해서 건강한 토론이 이루어진다.
   - 프로젝트 기간을 고려해서 기존 기획을 알맞게 수정했다.

## Problem 😢
   - 시간적 제한이 있기에 개발 시작을 서두르면 좋을 것 같다.
   - 의견을 묻기 전에 먼저 말했으면 좋겠다.
   - 스프린트 단위당 업무의 양과 난이도를 정확히 파악하지 못했다.
  
## Try 👍
   - 프로젝트 기획안을 추후에 변경되지 않게 확실하게 작성해야 한다.
   - 개발 관련 사소한 부분은 우선 넘어가고 작업을 시작하기로 했다.

<br>


![natepann_temp1487079666342-1395876642](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/a264f00c-1527-4882-bdd8-4747e80b275f)

6팀 파이팅!!!!

**Comment by JISOuO** [comment]
# 10팀😎

## Keep 
- 컨벤션을 꼼꼼하게 정해놔서 코드 짤때 보기가 좋다.
- 기획 아이디어를 잘내서 일처리가 빠르다.
- 가감없이 깔끔하게 피드백을 주고받아서 스스로 성장할 수 있다.
- 적극적으로 각자의 할 일을 계속 찾아서 진행이 빨랐다.

## Problem 
- 처음 진행하는 협업이어서 다른 포지션과 적절한 소통 빈도를 모르겠다.

## Try 
- 전역관리(서버+클라이언트) 라이브러리 잘써보기

![갱얼지](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/114739219/540c47e7-e2a8-4e55-829c-61101a845e86)

마지막 파트는 10팀이 점령하겠습니다+_+

**Comment by muyahho** [comment]
4팀 KPT 회고

Keep
 
코드작업에 어려움이나 문제가 있을때 바로 공유하고 쉐어해서 함께 해결을 잘 해나가고 있다.

 
회의 때 모두 적극적으로 참여하고 R&R 분배가 적절히 잘된 것 같다.

 
코드 작성 시, 다양한 방향성의 의견이 나온다.

 
프로젝트 진행시에 서로 배려해주는 건강한 분위기

Problem

 
다들 코드리뷰를 겁내는 것 같다.

 
소통의 오류가 있어 보다 명확한 소통하기

 
아직 코딩 실력이 완벽하지 않아서 swagger나 서버 쪽 문제인지 코드의 문제인지 파악하기 어렵고, 실제로 서버 문제가 몇 개 있었다.

 
PR 올릴 때 너무 오랫동안 작업한 걸 올려서 코드량이 방대해 팀원들 코드 리뷰가 힘들었다.

 
에러가 발생했을 때 복잡한 에러는 찾기가 어렵다

Try

 
코드 리뷰 시, 리뷰 할 것이 없으면 질문 하나씩은 꼭 해보기

 
PR을 작은 단위로 자주 올리도록 한다.

 
좀 더 작은 이슈라도 공유를 활성화하기

**Comment by bjiwon324** [comment]
# 5팀 🐼

## Keep 

- 팀회의가 체계적으로 잘 잡혀있습니다.
- 매일 팀회의록 작성 중입니다.
- 의사소통이 원활하게 잘 이루어집니다.

## Problem 

- 아직 적응 중이라 코드 리뷰가 조금 느립니다.
- 코드 리뷰를 적극적으로 하지 못 하는 것 같습니다.

## Try

- 빠른 코드 리뷰하기 
- 멘토님한테 질문 많이하기 


![happy-friday-dance_1](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148836373/cf207ba1-17d5-47a6-96b0-2f3dd6c98162)

**Comment by juncastle97** [comment]
# 9팀

## keep
- 미팅 회의록을 매일 작성하고 있다.
- 회의 참석률이 좋다.
- 컨벤션을 잘 정했다.
- 기획명세서를 자세히 작성했다.
## problem
- 처음해보는 협업이라 각자 영역에 대한 이해 부족으로 소통에 문제가 있었다.
- 개발 속도가 늦춰졌다.
## try
- 질문할 때 좀 더 구체적으로 질문하기!
- 서로에게 질문 많이 하기
- 마일스톤 확실하게 정해 개발 속도 향상시키기
- 대화 많이 하기!

**Comment by SeanKim-416** [comment]
# 11팀


![스크린샷 2024-03-11 오후 1 58 03](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/1c4a9512-c401-4bb9-a082-1a472dc28cc7)

<hr>


![스크린샷 2024-03-11 오후 1 58 17](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/e5367e84-f863-4115-bf6b-c00ce205e9ad)

<hr>

![스크린샷 2024-03-11 오후 1 58 37](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/21ddd464-aa2a-4436-8e24-99cea08e9e12)

**Comment by CheeseB** [comment]
# 3팀🦸🏼 

## Keep

### @designsoo 
- 항상 열정이 넘치는 팀 분위기가 너무 좋습니다!!
- 각자 의견을 잘 조율합니다.(다양한 의견도 많음)
- 전투적입니다
- 코드 리뷰 진행도 잘 되고 있어서 좋습니다!

### @grapefruit13 
- 겁없이 새로운 것에 도전하는 정신이 너무 좋습니다.
- 누가 시키지 않아도 주도적으로 행동합니다.
- 열심히 합니다.
- 서로의 의견을 나누고 결론을 도출하는 과정을 재밌게 해서 좋습니다.

### @ccwnc 
- 각자 맡은 부분을 알아서 잘 하고 있는 거 같아서 좋아요
- 의견을 말하고 결정을 내리는 과정이 매끄럽게 진행돼서 좋습니다

### @CheeseB 
- 새벽 거의 5시쯤까지 다들 함께있고 미니 멘토링까지 받았던 상황을 생각하면 정말 대단한거 같습니다 저희 팀ㅋㅋㅋㅋㅋ그만큼 다들 열정이 넘치셔서 좋아요!

## Problem

### @designsoo 
- 이유가 타당할때까지 의논합니다 ㅋㅋㅋ다들 멈추지 않습니다.

### @grapefruit13 
- 이번 주에 한 3번 정도 삼천포로 또 빠져버렸습니다.
- 바이오리듬이 망가지지 않았는지 걱정이 됩니다.
- 회의시간이 조금 길어질 때가 있습니다.

### @ccwnc 
- 올빼미가 된 거 같습니다

### @CheeseB 
- 잠은 잘 주무시면서 하면 좋겠습니다ㅋㅋㅋ

## Try

### @designsoo 
- 건강합시다!

### @grapefruit13 
- 그다지 중요하지 않은 것과 중요한 것을 더 잘 구분해서 시간을 할애하면 좋을 것 같습니다. 쉽지 않지만 계속 노력해봅시다😃
- 일찍 자고 일찍 일어날 수 있게 노력해보겠습니다.

### @ccwnc 
- 잠을 잘 잡시다
- 앞으로 몇주 더 남았는데 처음부터 시간을 너무 길게 쓰면 지치니까 적당한 시간을 딱 정해놓고 회의합시다

### @CheeseB 
- 안되면 더 시간을 쓰면 된다는 생각보다는(제 얘기입니다..), 주어진 시간 안에 최대한 다 끝낸다는 생각으로 빡집중해서 작업시간을 효율적으로 쓰면 잠자는 시간을 좀더 늘릴수 있지 않을까 싶습니다..!

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/43297823/714875d7-a7c9-4e0a-98ad-4ae1a1e03b74)

**Comment by rladngus133** [comment]
## 2팀 🦎
### Keep
- 활발한 코드리뷰가 이루어져서 다 같이 열심히하는 모습이 보기가 좋았습니다.
- 팀활동시 서스럼없이 궁금한점을 다 같이 질의응답하는 모습이 좋았고 공부가 되었습니다.
- 컨벤션을 초기에 잘 구성하여서 꾸준히 잘 지켜지고있어 더 편했습니다.
- PR, 이슈등 GitHub의 기능들을 잘 활용하여 체계적으로 작업이 가능하여 좋았습니다.

### Problem
- 머지가 조금더 신속하게 이루어지면 좋겠다.

### Try
- 코드 리뷰에대한 답변과 반영을 빠르게하고 이상이 없을시 빠르게 머지하자 ! 

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/124851297/55c846b9-637c-49e3-bff6-73bb233316e1)

**Comment by MoonPillGyeom** [comment]
## 8팀
## Keep
- 스크럼 시간에 디자인, 백엔드, 프엔 분들이 모여서 일정회의와 논의사항을 잘 공유하고 있어요.
- 작업속도를 잘 유지하고 있는 것 같습니다. 
- 참여도 진행도 현재까지 원활한 거 같습니다
- 시작부터 지금까지 작업 페이스가 원활하게 흘러가고 있는 것 같습니다.
## Problem
- 아직 없는 것 같습니다... 
- 소통/코드리뷰 부족
 - 자잘한 코드 컨벤션에 대한 문제
 - 회의 시스템 / 기록 부재
## Try
- 서로 코드에 대한 이해를 높이기 위한 시간을 갖는 것도 좋을 것 같습니다.
- 실시간으로 모여서 서로 논의해보는 시간을 가져도 괜찮을 것 같습니다.
- 시간을 정해서 각자 다른 사람들의 코드를 보고 리뷰를 다는 시간을 가지면 서로의 코드를 이해하는 데 더 도움이 될 것 같습니다.
- 회의 시간에 한 번 더 컨벤션에 관한 부분 짚고 넘어가면 좋을 것 같습니다.
- 데일리 스크럼 / 팀 회의 템플릿을 만들어 템플릿대로 회의가 진행되면 좋을 것 같습니다.

**Comment by 00TaciTa00** [comment]
# 7팀
- BE와 DE 포함하여 익명으로 진행되었습니다.

## Keep
- 같이 디스코드 서버 만들어서 작업하는 거 너무 좋았어요
- 서로 의견 제시하면서 기획안을 성공적으로 제출할 수 있어서 좋았습니다.
- 적극적으로 의견을 PR하고 서로의 관점을 이해하고 수용하는 태도
- 모르는 것을 정확하게 말하고 개선해가려는 분위기
- 질문하는게 어렵지 않은 분위기
- 회의 내용 잘 정리해 주는 점이 좋아요
- 회의할 때 잡담이 거의 없는 거 좋아요

## Problem
- 조금 진도가 느린 거 같아요
- 걱정이 너무 많은 거 같아요
- 디자이너분이 바쁘신게 염려가 됩니다.

## Try
- 조금만 더 후딱후딱 해보죠
- 자신감 있게 진행해도 좋을 거 같아요
- 회의 내용 및 시간에 대해 더 명확하게 진행되었으면 좋겠어요
